app/src/main/java/ch/epfl/culturequest/ui/map/MapsFragment.java

```diff
@@ -73,24 +97,74 @@ public class MapsFragment extends Fragment {
         @Override
         public void onMapReady(GoogleMap googleMap) {
             mMap = googleMap;
+            drawPositionMarker(viewModel.getCurrentLocation().getValue());
             mMap.setMapStyle(MapStyleOptions.loadRawResourceStyle(getContext(), R.raw.maps_style_json));
             getLocationPermission();
             mMap.moveCamera(CameraUpdateFactory
                     .newLatLngZoom(viewModel.getCurrentLocation().getValue(), DEFAULT_ZOOM)); // Set to Default location anyway
         }
     };
 
-    public void getMarkers(){
+
+    private void drawPositionMarker(LatLng latestLocation){
+        frame = mMap.addMarker(new MarkerOptions().zIndex(10000f).position(latestLocation).icon(BitmapDescriptorFactory.fromBitmap(Bitmap.createScaledBitmap(BitmapFactory.decodeResource(getResources(), R.drawable.map_icon_frame), 75, 75, false))));
+        if(profilePicture != null) {
+            profileMarker = mMap.addMarker(new MarkerOptions().zIndex(10001f).icon(BitmapDescriptorFactory.fromBitmap(Bitmap.createScaledBitmap(profilePicture, 70, 70, false))).position(latestLocation));
+        }
+    }
+
+    // Method to change the profile picture from square to round
+    private static Bitmap getCircularBitmap(Bitmap bitmap) {
+        Bitmap output;
```

Comment:
In the next sprint we will save the profile pictures as rounds so maybe you will not need this function anymore at the end of next sprint. But for now keep it like that :)

Reply:
Perfect, thanks for the heads up :)